nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,913 @@
+                                      bg_img=bg_img,
+                                      display_mode=display_mode,
+                                      )
+    all_components = _make_stat_maps_contrast_clusters(
+            stat_img=statistical_maps,
+            contrasts_plots=contrast_plots,
+            threshold=threshold,
+            alpha=alpha,
+            cluster_threshold=cluster_threshold,
+            height_control=height_control,
+            min_distance=min_distance,
+            bg_img=bg_img,
+            display_mode=display_mode,
+            plot_type=plot_type,
+            )
+    all_components_text = '\n'.join(all_components)
+    report_values = {'page_title': escape(page_title),
+                     'page_heading_1': page_heading_1,
+                     'page_heading_2': page_heading_2,
+                     'model_attributes': model_attributes_html,
+                     'all_contrasts_with_plots': ''.join(
+                         contrast_plots.values()),
+                     'design_matrices': html_design_matrices,
+                     'roi_plot': roi_plot_html_code,
+                     'component': all_components_text,
+                     }
+    report_text = report_template.safe_substitute(**report_values)
+    report = HTMLDocument(report_text)
+    report.width = nb_width  # better visual experience in Jupyter Notebooks.
+    report.height = nb_height
+    return report
+
+
+def _coerce_to_dict(input_arg):
+    """ Constructs a dict from the provided arg.
+    
+    If input_arg is:
+      dict then returns it unchanged.
+      
+      string or collection of Strings or Sequence[int],
+      returns a dict {str(value): value, ...}
+    
+    Parameters
+    ----------
+    input_arg: String or Collection[str or Int or Sequence[Int]]
+     or Dict[str, str or np.array]
+        Can be of the form:
+         'string'
+         ['string_1', 'string_2', ...]
+         list/array
+         [list/array_1, list/array_2, ...]
+         {'string_1': list/array1, ...}
+    
+    Returns
+    -------
+    Dict[str, np.array or str]
+
+    """
+    if not isinstance(input_arg, dict):
+        excluded_types = (list, tuple, np.ndarray, str)
+        if isinstance(input_arg, excluded_types):
+            if not isinstance(input_arg[0], excluded_types):
+                input_arg = [input_arg]
+        input_arg = [input_arg] if isinstance(input_arg, str) else input_arg
+        input_arg = {str(contrast_): contrast_ for contrast_ in input_arg}
+    return input_arg
+
+
+def plot_to_svg(plot):
+    """
+    Creates an SVG image as a data URL
+    from a Matplotlib Axes or Figure object.
+
+    Parameters
+    ----------
+    plot: Matplotlib Axes or Figure object
+        Contains the plot information.
+
+    Returns
+    -------
+    url_plot_svg: String
+        SVG Image Data URL
+    """
+    with io.BytesIO() as buffer:
+        try:
+            plot.figure.savefig(buffer, format='svg')
```

Comment:
maybe a comment to say that this is for nilearn displays

Reply:
I don't understand this. Why is this for Nilearn displays?

Reply:
Ah I am guessing you mean its for one of Nilearn's display objects? Its not, atleast not intentionally.
The try block is for  for axes objects, the except block for figure objects.

Reply:
sorry, you can ignore this comment